### PR TITLE
Initial tcp server with line protocol implementation

### PR DESCRIPTION
**Pull request by Deitrl**

### CODE REVIEW

File: carbon/src/bin/carbon-server.rs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+fn run(args: Args) -> Result<(), Error> {
+    if args.generate {
+        Settings::generate(args.config.unwrap())?;
+        exit(1);
+    }
+
+    let settings = Settings::new(args.config)?;
+    println!("{:?}", settings);
```
Comment (andy128k) [inline carbon/src/bin/carbon-server.rs]: ?

File: carbon/src/bin/carbon-server.rs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+    let addr = format!("{0}:{1}", &settings.tcp.host, settings.tcp.port).parse()?;
+    let listener = TcpListener::bind(&addr)?;
+    let path = Arc::new(settings.db_path);
+    let config = Arc::new(settings.whisper);
```
Comment (andy128k) [inline carbon/src/bin/carbon-server.rs]: looks like those two are always together. Probably it's better to have `let settings_arc = Arc::new(settings)`.

File: carbon/src/bin/carbon-server.rs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                    .duration_since(UNIX_EPOCH)
+                    .unwrap()
+                    .as_secs() as u32;
+                line_update(&line, &p, &conf, now)
+                    .map_err(|e| io::Error::new(io::ErrorKind::Other, e.to_string()))?;
+                Ok(())
+            })
+        })
+        .map_err(|err| {
+            println!("accept error = {:?}", err);
```
Comment (andy128k) [inline carbon/src/bin/carbon-server.rs]: eprintln ?

File: carbon/src/lib.rs [hunk 4]
```diff
@@ -110,21 +120,80 @@ impl FromStr for MetricPath {
...
 impl From<PathBuf> for MetricPath {
-    fn from(metric_path: PathBuf) -> MetricPath {
+    fn from(metric_path: PathBuf) -> Self {
         MetricPath(metric_path)
     }
 }
 
+pub fn line_update(
+    message: &str,
+    dir: &PathBuf,
```
Comment (andy128k) [inline carbon/src/lib.rs]: &Path

Comment (AnderEnder) [reply carbon/src/lib.rs]: implemented as `AsRef<Path>`

**Pull request merged**